### ISSUE TITLE
1.23.43 doesn't work, register request 400 Bad Request/403 Forbidden error

### ISSUE THREAD

**Issue by mnordhoff** [tailscale/tailscale#4131]
### What is the issue?

I've had Tailscale unstable installed on Ubuntu 20.04 (amd64) for a while. Upgrading from 1.23.30 to 1.23.43... doesn't work. Tailscale logs repetitive errors, as below, and I can't send traffic through it. Presumably it's unable to talk to control.

I downgraded to 1.23.30 again and it works.

The install is old enough that it still uses login.tailscale.com as the control server.

(This is a Linode VPS. The network doesn't do NAT or interception.)

Log snippet:

```
Mar 10 02:49:49 ultimate tailscaled[2441296]: Start
Mar 10 02:49:49 ultimate tailscaled[2441296]: using backend prefs for "_daemon": Prefs{ra=false dns=true want=true routes=[] nf=on url="https://login.tailscale.com" Persist{lm=[rw+Mv], o=, n=[M5e4C] u="mnordhoff@gmail.com"}}
Mar 10 02:49:49 ultimate tailscaled[2441296]: Backend: logs: be:8a765d4bf3a9b7e3398c745fce93d3aa0729a7c712724315313045c45b8965ae fe:
Mar 10 02:49:49 ultimate tailscaled[2441296]: control: client.Login(false, 0)
Mar 10 02:49:49 ultimate tailscaled[2441296]: control: doLogin(regen=false, hasUrl=false)
Mar 10 02:49:49 ultimate tailscaled[2441296]: control: control server key  from https://login.tailscale.com
Mar 10 02:49:49 ultimate tailscaled[2441296]: control: RegisterReq: onode= node=[M5e4C] fup=false
Mar 10 02:49:50 ultimate tailscaled[2441296]: Received error: register request: Post "https://login.tailscale.com/machine/register": all connection attempts failed (HTTP: unexpected HTTP response: 400 Bad Request, HTTPS: unexpected HTTP response: 403 Forbidden)
Mar 10 02:49:50 ultimate tailscaled[2441296]: control: doLogin(regen=false, hasUrl=false)
Mar 10 02:49:50 ultimate tailscaled[2441296]: control: RegisterReq: onode= node=[M5e4C] fup=false
Mar 10 02:49:50 ultimate tailscaled[2441296]: Received error: register request: Post "https://login.tailscale.com/machine/register": all connection attempts failed (HTTP: unexpected HTTP response: 400 Bad Request, HTTPS: unexpected HTTP response: 403 Forbidden)
```

The bugreport below is from the second time I upgraded to 1.23.43.

I have another one, BUG-8a765d4bf3a9b7e3398c745fce93d3aa0729a7c712724315313045c45b8965ae-20220310025248Z-fe723e94950f90e8, after downgrading to 1.23.30 again.

### Steps to reproduce

I don't know useful steps to reproduce.

All I did was upgrade from 1.23.30 to 1.23.43.

You may have to configure it to use login.tailscale.com, or there may be other conditions.

### OS

Linux

### OS version

Ubuntu 20.04

### Tailscale version

1.23.43

### Bug report

BUG-8a765d4bf3a9b7e3398c745fce93d3aa0729a7c712724315313045c45b8965ae-20220310025043Z-6b4a2a6c21569c8a

**Comment by danderson** [comment]
Thanks for the report! We're changing the control protocol around, and you hit a thing that managed to escape us during testing. The fix is merged, we'll probably roll a new unstable tomorrow to get it fixed.

Thanks for braving the unstable track!

**Comment by mnordhoff** [comment]
:)

Thank you for the quick response.

The unstable track is usually awfully stable, really. :D

**Comment by bradfitz** [comment]
1.23.47 is out now which should fix it for you. Let us know and we'll investigate if it doesn't help.

**Comment by mnordhoff** [comment]
Verified fixed in 1.23.47. :)
